### DISCUSSION TITLE
Idea for improving the selection of spoken language

### DISCUSSION THREAD

**Discussion by mrburns-42** [LeoKlaus/plappa#91] [Feature Requests]
@LeoKlaus Thank you for implementing the option to filter by the spoken language of an audiobook, that already helps.

As a suggestion for improvement, I would like to be able to select the language via a menu in addition to the search. For example, the Language menu item could be placed here:

![IMG_6748](https://github.com/LeoKlaus/plappa/assets/786982/c6e2147c-c31d-4382-9d78-eaab764633ac)

The advantage I see is that I can select a language and then also use the search function. It is also more intuitive for the user and the user can see in which languages audiobooks are available

Since Jellyfin and Emby do not give out any information about the spoken language as far as I know, one idea would be that the menu item is only displayed if you are connected to an ABS server.

**Comment by LeoKlaus** [comment]
I like that idea. I don't think the ABS api offers any list of available languages or something like that, so I would have to implement some internal tracking to display a list.

**Reply by mrburns-42** [reply]
I've just had a look at the API documentation, at least all languages can be transferred from the server:

https://api.audiobookshelf.org/#library-filter-data

And if I see correctly, you can also filter by language. However, I've only had a very superficial look at it all.

https://api.audiobookshelf.org/#filtering

**Reply by LeoKlaus** [reply]
That's perfect, good catch!
